python/dask_cudf/dask_cudf/expr/_collection.py

```diff
@@ -81,6 +82,22 @@ def from_dict(cls, *args, **kwargs):
         with config.set({"dataframe.backend": "cudf"}):
             return DXDataFrame.from_dict(*args, **kwargs)
 
+    def sort_values(
+        self,
+        by,
+        **kwargs,
+    ):
+        # Check if first column is categorical and raise if it is
+        check_by = by[0] if isinstance(by, list) else by
```

Comment:
Does this only seg-fault when the first sort column is categorical, or could it be any column in sort?

Reply:
It only needs to raise if the first column is categorical, otherwise we don't use the problematic quantile/percentile logic to calculate divisions.

Reply:
In that case, want to mention that in the comment here? Not sure if there's a relevant Dask issue we could link to:

```suggestion
        # raise to avoid computing quantiles on categorical columns here
        check_by = by[0] if isinstance(by, list) else by
```

Reply:
How about:

```python
        # Raise if the first column is categorical, otherwise the
        # upstream divisions logic may produce errors
        # (See: https://github.com/rapidsai/cudf/issues/11795)
```

Reply:
Yup that looks great 🙂 

```suggestion
        # Raise if the first column is categorical, otherwise the
        # upstream divisions logic may produce errors
        # (See: https://github.com/rapidsai/cudf/issues/11795)
        check_by = by[0] if isinstance(by, list) else by
```